### PR TITLE
Activity Log: migrate updates away from data getters

### PR DESCRIPTION
**Pull request by jsnajdr**
This is a followup to #60393 which migrated the Site Alerts data getter to React Query. Now we're migrating the POST requests (mutations) that update a theme or Core from a data getter to a plain `wpcom.req.post` request.

It quite a major rewrite of the entire component, here's some info how to best understand it:

The component is getting `plugins`, `themes` and `core` props with the list of available updates, and is getting them from Redux (plugins) and React Query site alerts (themes and core). Before this PR, there were also `pluginsWithUpdate` derived props that added an `updateStatus` field with the status of the update.

The `updateStatus` field was used only for two purposes:
- a `componentDidUpdate` method was detecting changes in `updateStatus` and then showed notices on success or error
- the queue processing code (`continueQueue`) started processing another item only when no other item was updating (`isItemUpdating` helper that checks for `inProgress` status)

This PR replaces that with:
- `then` and `catch` handlers on the promise returned by `updateSingle`
- `this.state.itemUpdating` boolean flag that's set before and after `updateSingle`

The `*WithUpdate` props and their logic is now no longer needed and can completely disappear: we only need the `plugins`/`themes`/`core` props.

**How to test:**
Create a Jurassic Ninja site with a WP Downgrade plugin and put it into state where there are pending Core, plugins and theme updates:

Screenshot 2022-01-24 at 10 18 15

Now click the "Update" buttons and verify that the updates proceeded successfully.

There's one caveat with a Core update: the WP Downgrade plugin registers some filters that pretend that the current Core version is the latest one. A Core update with the active plugin will therefore return a "WordPress is at the latest version" error. You'll need to deactivate the WP Downgrade plugin to make the Core update succeed.

### CODE REVIEW

File: client/my-sites/activity/activity-log-tasklist/index.jsx [hunk 1]
```diff
@@ -247,6 +216,37 @@ class ActivityLogTasklist extends Component {
...
+
+		updateSingle( item, siteId )
+			.then( () => {
+				showSuccessNotice(
+					translate( 'Successfully updated %(item)s on %(siteName)s.', {
+						args: { item: decodeEntities( item.name ), siteName },
+					} ),
+					{
+						id: `alitemupdate-${ item.slug }`,
+						duration: 3000,
```
Comment (tyxla) [inline client/my-sites/activity/activity-log-tasklist/index.jsx]: I know this was here before, but 3 seconds might be a bit quick to read this sometimes. And it's not the shortest message if the plugin and site names are longer. Maybe we should change to 5s at least?

File: client/my-sites/activity/activity-log-tasklist/index.jsx [hunk 2]
```diff
@@ -2,7 +2,6 @@
 
 import { Card } from '@automattic/components';
 import { localize } from 'i18n-calypso';
-import { isEmpty, get, includes, find } from 'lodash';
```
Comment (tyxla) [inline client/my-sites/activity/activity-log-tasklist/index.jsx:5]: Thanks for removing some more Lodash 🙇‍♂️

**Review by tyxla — APPROVED** [review]
Both single button updates and all updates at once worked well for me (tested them separately on 2 Jetpack sites). ✅ 

Code also looks good. 👍 

Loving all the cleanup and the simplicity of the new version! 🚢

File: client/my-sites/activity/activity-log-tasklist/index.jsx (hunk 1 above)
Comment (jsnajdr) [reply client/my-sites/activity/activity-log-tasklist/index.jsx]: Changed this to the default 5s duration :+1:

**Pull request merged**